impl.go

```diff
@@ -26,6 +26,34 @@ var (
 	flagRecvPkg  = flag.String("recvpkg", "", "package name of the receiver")
 )
 
+func parseTypeParams(in string) (string, []string, error) {
```

Comment:
Some docs here would be nice. I can't guess from the name what the arg or the first two results are.

I think this would be simpler and easier to read if you used strings.Cut. E.g

```go
func parseTypeParams(in string) (string, []string, error) {
  id, rest, ok := strings.Cut(in, "[")
  if !ok {
    return in, nil, nil
  }
  params, rest, ok := strings.Cut(rest, "]")
  if !ok {
    // ...
  }
  if rest != "" {
    // ...
  }
  // ...
}
```

Reply:
Fair enough on both counts!